src/js/serial.js

```diff
@@ -20,6 +20,8 @@ const serial = {
         const testUrl = path.match(/^tcp:\/\/([A-Za-z0-9\.-]+)(?:\:(\d+))?$/);
         if (testUrl) {
             self.connectTcp(testUrl[1], testUrl[2], options, callback);
+        } else if(options.connectionType === 'virtual'){
```

Comment:
```suggestion
        } else if (options.connectionType === 'virtual') {
```
You could have used `path` parameter to set it to `virtual` connectionType to test here?